src/aggregate/aggregate_request.c

```diff
@@ -931,10 +924,9 @@ static ResultProcessor *getGroupRP(AREQ *req, PLN_GroupStep *gstp, ResultProcess
 static ResultProcessor *getVecSimRP(AREQ *req, AGGPlan *pln, ResultProcessor *up, QueryError *status) {
   RLookup *lk = AGPLN_GetLookup(pln, NULL, AGPLN_GETLOOKUP_LAST);
   char **scoreFields = req->ast.vecScoreFieldNames;
-  size_t len = array_len(scoreFields);
-  RLookupKey **keys = rm_calloc(len, sizeof(*keys));
-
-  for (size_t i = 0; i < len; i++) {
+  size_t nScoreFields = array_len(scoreFields);
+  RLookupKey **keys = rm_calloc(nScoreFields, sizeof(*keys));
```

Comment:
We can allocate it on the stack no? Or even use the array_on_stack you implemented on gears?

Reply:
Forget...